### PR TITLE
Update intro tutorial (#1116)

### LINKED ISSUE #1116: Update `ModelChain.run_model` documentation
Related to [this thread in Google Groups about difficulties when trying to go from object-oriented to procedural API](https://groups.google.com/g/pvlib-python/c/I5U0owXvyc8).

So basically `.run_model()` does a great job at simplifying the required code in order to run a simulation, but it does a lot of "magic" behind the scenes and it is not easy to follow all the logic just looking at the source code, so maybe some documentation could be added to improve this. How much or where to add it? (docstring, an example under `docs/`...) I don't know. :innocent: But if you tell me, I'd be happy to contribute a PR.

For this particular case, it was not clear to me that `.run_model()` was, when calculating the solar position apparent zenith, using the TMY air temperature series and the calculated pressure at the given altitude.

Maybe I could update [the example in the documentation](https://pvlib-python.readthedocs.io/en/stable/introtutorial.html) to make the results match exactly? At least this way there is an example on how to use both paradigms to get the same results.

### PR DESCRIPTION
**Pull request by Peque**
- [x] Closes #1116
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [ ] ~Tests added~ (N/A)
 - [ ] ~Updates entries to [`docs/sphinx/source/api.rst`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/api.rst) for API changes.~ (N/A)
 - [ ] ~Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).~ (N/A)
 - [ ] ~New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.~ (N/A)
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

This MR updates the example code in the intro tutorial. A couple of things to note:

- The procedural and OO results match exactly, as expected
- Instead of clearsky, we are using TMY data from PVGIS, highlighting the use of that `iotools` function

Also, I could add some transformations to the downloaded TMY data to avoid possible confusion to new users, in particular:

- Replace TMY index year with 2019, so in case the user plots the energy series, they see a continuous year
- Convert TMY index to local timezone, so in case the user plots the hourly energy, they see a more expected curve

### CODE REVIEW

**Comment by Peque** [comment]
Reviewing my own MR, yeah, maybe I could just use `sapm_effective_irradiance()`, which simplifies the code for this intro tutorial. If they user wants to try with other modules not from the CEC dabase, they'll probably find out what it needs to be done?

File: docs/sphinx/source/introtutorial.rst [hunk 1]
```diff
@@ -48,9 +47,29 @@ configuration at a handful of sites listed below.
...
 
-    # specify constant ambient air temp and wind for simplicity
-    temp_air = 20
-    wind_speed = 0
+
+Let's download the typical meteorological year weather data from PVGIS, which
+includes irradiation, temperature and wind speed.  Note that PVGIS uses
+different naming conventions, so it is required to rename the weather data
+variables before using them. PVGIS weather data is already UTC-localized, so
+conversion to local timezone is optional.
```
Comment (wholmgren) [inline docs/sphinx/source/introtutorial.rst:55]: The text itself is pretty neutral, but I don't want to imply that we favor PVGIS. I'm wondering if we need a sentence here like ``See :ref:`iotools` for other weather data sources.`` or maybe even something more explicit.

**Review by wholmgren — COMMENTED** [review]
Thanks @Peque this is great. One comment below and can you also add a note to the 0.9 whats new file?

File: docs/sphinx/source/introtutorial.rst (hunk 1 above)
Comment (Peque) [reply docs/sphinx/source/introtutorial.rst]: Pushed a [fixup commit with a little change](https://github.com/pvlib/pvlib-python/pull/1144/commits/c86a71c5385db21487dd3d2e6a175df3815a6298). Feel free to rephrase it as you'd like if you still prefer some other way to say it. :blush:

**Review by wholmgren — APPROVED** [review]

**Review by cwhanse — APPROVED** [review]
Thanks @Peque .

My only comment is outside the scope of this tutorial: it is unfortunate that the key for getting `'sandia'` inverter parameters using `retrieve_sam` is `'cecinverter'`.  "sandia", "SAM" and "cec" all appear in this line of code; no surprise that novices get confused.

**Comment by Peque** [comment]
By the way, just in case you missed my comment above, I could add some transformations to the downloaded TMY data to avoid possible confusion to new users, in particular:

- Replace TMY index year with 2019, so in case the user plots the energy series, they see a continuous year (PVGIS assigns a datetime with the year it got the typical meteorological month data from)
- Convert TMY index to local timezone, so in case the user plots the hourly energy, they see a more expected curve

Like this:

```diff
--- a/docs/sphinx/source/introtutorial.rst
+++ b/docs/sphinx/source/introtutorial.rst
@@ -71,7 +71,11 @@ includes irradiation, temperature and wind speed.
         latitude, longitude, name, altitude, timezone = location
         weather = pvlib.iotools.get_pvgis_tmy(latitude, longitude)[0]
         weather = weather.rename(columns=variables_translation)
+        weather.index = weather.index.map(lambda t: t.replace(year=2019))
         weather.index.name = "utc_time"
+        # Optional conversion to local timezone
+        weather.index = weather.index.tz_convert(timezone)
+        weather.index.name = "local_time"
         tmys.append(weather)
```

Or would you rather leave it as UTC and as isolated months from different years?

**Comment by wholmgren** [comment]
let's keep it as is - the "intro example" can't do everything. thanks @Peque!

**Comment by EricDuminil** [comment]
Excellent, thanks for the work!

It might be a good time to correct the inaccurate coordinates in `introtutorial.rst`.
Right now, San Francisco is in Reno and Berlin is almost in France.

If you're interested:

```diff
     import pandas as pd
     import matplotlib.pyplot as plt
 
-    # very approximate
     # latitude, longitude, name, altitude, timezone
     coordinates = [
-        (30, -110, 'Tucson', 700, 'Etc/GMT+7'),
-        (35, -105, 'Albuquerque', 1500, 'Etc/GMT+7'),
-        (40, -120, 'San Francisco', 10, 'Etc/GMT+8'),
-        (50, 10, 'Berlin', 34, 'Etc/GMT-1'),
+        (32.2, -111.0, 'Tucson', 700, 'Etc/GMT+7'),
+        (35.1, -106.6, 'Albuquerque', 1500, 'Etc/GMT+7'),
+        (37.8, -122.4, 'San Francisco', 10, 'Etc/GMT+8'),
+        (52.5, 13.4, 'Berlin', 34, 'Etc/GMT-1'),
     ]
```
Here's the commit:
https://github.com/EricDuminil/pvlib-python/commit/7abbfc8e7d9c4a59ee1b937948cb4eefeaab1801

**Pull request merged**